[docs] Add CodeSandbox/Stackblitz to the rest of the templates

siriwatknp opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

**Preview**:
- [Dashboard](https://deploy-preview-43708--material-ui.netlify.app/material-ui/getting-started/templates/dashboard/)
- [Marketing page](https://deploy-preview-43708--material-ui.netlify.app/material-ui/getting-started/templates/marketing-page/)
- [Checkout](https://deploy-preview-43708--material-ui.netlify.app/material-ui/getting-started/templates/checkout/)
- [Sign in side](https://deploy-preview-43708--material-ui.netlify.app/material-ui/getting-started/templates/sign-in-side/)
- [Sign up](https://deploy-preview-43708--material-ui.netlify.app/material-ui/getting-started/templates/sign-up/)
- [Blog](https://deploy-preview-43708--material-ui.netlify.app/material-ui/getting-started/templates/blog/)

Follow up PR from #43604

- Mostly deleting duplicated theme files in each template (switch to use the shared theme)
🫣 <img width="150" alt="image" src="https://github.com/user-attachments/assets/07523a6e-7470-4cbc-9bc2-1cfbf4244a9e">
- Fixed minor CSS after switching to shared TemplateFrame
- Refactor `theme.palette.*` to use `theme.vars.palette.*`

---

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

zanivan commented:
@siriwatknp, these are the issues I found during my initial review. I won't dive too deep into visual tweaks for the templates, like the box-shadow on the marketing page appbar, since that's not the focus of this PR. I've kept the feedback primarily focused on layout and interaction issues based on `TemplateFrame`:

- There's an issue on the mobile layout of the Marketing-page, Sign-in, and Checkout templates:
<img width="220" alt="Screenshot 2024-09-11 at 08 55 53" src="https://github.com/user-attachments/assets/1d2416f7-5fb9-491b-a96a-fdcf48a842ca"><img width="200" alt="Screenshot 2024-09-11 at 08 56 13" src="https://github.com/user-attachments/assets/b7c59a1c-9ca3-4443-bed5-630f15bca087"><img width="220" alt="Screenshot 2024-09-11 at 08 58 59" src="https://github.com/user-attachments/assets/c88c03d5-bec1-4097-936a-5398a2b7cb38"><img width="200" alt="Screenshot 2024-09-11 at 08 57 54" src="https://github.com/user-attachments/assets/e1403009-23e7-440f-abf0-3537f16acb99">

- The `AppBar` on Blog template is not centralized:  <img width="1266" alt="Screenshot 2024-09-11 at 09 01 26" src="https://github.com/user-attachments/assets/49500665-8dd2-4bba-871d-6e0ed562feb3">

- The drawer in the Blog/Marketing page template is placed behind the `TemplateFrame`: <img width="427" alt="Screenshot 2024-09-11 at 09 00 37" src="https://github.com/user-attachments/assets/197c343a-fd03-4456-98f1-608c07d2f84f">

- The same issue happens in the mobile drawer of Checkout template: <img width="374" alt="Screenshot 2024-09-11 at 08 58 16" src="https://github.com/user-attachments/assets/cba1ddd8-3059-45a2-9bd5-ea4f25a08c9e"><img width="376" alt="Screenshot 2024-09-11 at 08 58 10" src="https://github.com/user-attachments/assets/13cab10e-fcbe-4be9-a767-b9ecd37942ff">

- There's a vertical scroll in the `Checkout` template:

https://github.com/user-attachments/assets/671b2405-3d39-4a4a-85ed-3875cdefdea9


- There's an arrow icon behind the color mode select (same as in [this comment](https://github.com/mui/material-ui/pull/43604#issuecomment-2343304086)): <img width="375" alt="Screenshot 2024-09-11 at 08 58 59" src="https://github.com/user-attachments/assets/c88c03d5-bec1-4097-936a-5398a2b7cb38">

siriwatknp commented:
Updates:

1. Make the theme selector smaller in mobile viewport

<img width="368" alt="image" src="https://github.com/user-attachments/assets/accf3649-bef7-4d54-bc78-9a306709288a">

2. Addressed all of the comments

DiegoAndai commented:
### Question

> Mostly deleting duplicated theme files in each template (switch to use the shared theme)

I thought the idea was to have duplicate theme files so each template would have all the code. This way users can clone one template without having to navigate to the shared theme. Is this no longer the case? @zanivan 

### Some issues found

**1. Checkout mobile dark mode**
<img width="300" alt="Screenshot 2024-09-12 at 09 45 13" src="https://github.com/user-attachments/assets/7039df82-1a52-4b2d-8349-13739e508d2a">

**2. Blog images not showing up**
<img width="300" alt="Screenshot 2024-09-12 at 09 44 06" src="https://github.com/user-attachments/assets/10e5bd63-126b-4f06-91a4-f1f96a2fab75">

**3. Stackblitx only: Sessions graph width not working as expected**

https://github.com/user-attachments/assets/5aef097d-ca1a-4037-9b85-3c47ec28d5da

zanivan commented:
> I thought the idea was to have duplicate theme files so each template would have all the code. This way users can clone one template without having to navigate to the shared theme. Is this no longer the case? @zanivan

This was the idea, as the only way users could get the templates before was by copying the folder from the project. However, integrating with CodeSandbox/StackBlitz solves this issue, since @siriwatknp packaged the shared theme within the template files when opened on either platform

zanivan commented:
Some bugs/issues I noticed while reviewing it again:

- The TemplateFrame toolbar sometimes gets stuck when scrolling the dashboard template: <img width="800" alt="Screenshot 2024-09-12 at 10 41 10" src="https://github.com/user-attachments/assets/52195892-043b-431c-b08b-057d398bf726">

- The AppBar from Blog and Marketing-page templates is not centralized: <img width="800" alt="Screenshot 2024-09-12 at 10 44 22" src="https://github.com/user-attachments/assets/d48338fe-d1f7-45e8-9d6a-39b77869b8e9">

- There's an issue in all templates in both StackBlitz and CodeSandbox regarding the dark/light theme:
<img width="400" alt="Screenshot 2024-09-12 at 10 42 02" src="https://github.com/user-attachments/assets/a0f7650a-d6d5-4c94-88f8-ed1ff98a6005">
<img width="400" alt="Screenshot 2024-09-12 at 10 43 52" src="https://github.com/user-attachments/assets/6e7a0805-1445-455c-9038-e2592f61de49">
<img width="400" alt="Screenshot 2024-09-12 at 10 44 49" src="https://github.com/user-attachments/assets/4ae897e8-5e96-49b0-b750-12aa16dcfd7c">
<img width="400" alt="Screenshot 2024-09-12 at 10 44 59" src="https://github.com/user-attachments/assets/1cff362e-5efc-4cc1-99c8-f5f3eb6c3199">

---

Just a disclaimer, there are other small UI improvements since we changed the way the `TemplateFrame` works, and some other adjustments, that can be addressed in a future PR.

zanivan commented:
> We have this new and improved way to use the templates with the sandboxes, and we should promote it, but it might not be best to "remove" the old way right away. There's no rush; we can wait and see how users react to this new system, and if it sticks, we can remove the old one (duplicated themes).

No strong opinion on this. Maybe we can wait until the feature to copy the theme to the clipboard is ready before removing the theme folders from the templates

siriwatknp commented:
> I think we could be a bit more cautious: We have this new and improved way to use the templates with the sandboxes, and we should promote it, but it might not be best to "remove" the old way right away

Thanks for pointing out. Let me clarify the removal of the duplicate theme. Using the shared-theme import has more benefits in several ways:

- Better DX, updating the shared theme immediately update all the templates, no longer need to run the script every time you change the code.
- Reduced complexity, no script, no CI check. Faster feedback loop, if you forget to run the script, you need to wait for more 15-20 minutes until the CIs are green.
- The dashboard
- Almost 20k lines are removed, the repo size is already huge (> 120MB). If we could optimize it, we should do it.

siriwatknp commented:
> * There's an issue in all templates in both StackBlitz and CodeSandbox regarding the dark/light theme:

This is a regression from https://github.com/mui/material-ui/pull/43632 due to `<StyledEngineProvider injectFirst>`

siriwatknp commented:
> * The AppBar from Blog and Marketing-page templates is not centralized:

Can you explain what not centralized?

siriwatknp commented:
@zanivan @DiegoAndai The dark mode bug that you see will be fixed by https://github.com/mui/material-ui/pull/43739.

zanivan commented:
> Can you explain what not centralized?

I don't know exactly if it's the `AppBar` or if it's the content, but there's a difference on the left/right alignment:

<img width="1183" alt="Screenshot 2024-09-13 at 09 26 27" src="https://github.com/user-attachments/assets/f1529460-3dcd-4702-bbb7-358f9b318246">

DiegoAndai commented:
> With all the above reasons, I don't see why we should keep the duplicated theme.

I understand the reasons, and I fully agree that we should do it at some point 👍🏼 my question was, "Are we ready to remove the previous DX?" which @zanivan is on board with, so let's do it.

Then my point becomes that we need to:
1. Update/remove the usage instructions on each template folder: https://github.com/mui/material-ui/tree/v6.1.0/docs/data/material/getting-started/templates/dashboard#usage, as not all files will be available in the templates' folders.
2. Remove https://github.com/mui/material-ui/blob/master/docs/scripts/updateTemplatesTheme.ts
3. Remove https://github.com/mui/material-ui/blob/master/package.json#L42

siriwatknp commented:
> 1. Update/remove the usage instructions on each template folder

Good point, did not know they exists.

Update all the readme to include copying `shared-theme` and remove the script.

siriwatknp commented:
@zanivan I could not see the difference. Is it the dashboard template?

<img width="898" alt="image" src="https://github.com/user-attachments/assets/a3783141-1bc4-49e0-9b35-18278142af60">

zanivan commented:
> @zanivan I could not see the difference. Is it the dashboard template?

@siriwatknp It's happening on `Marketing-page` and `Blog`, and I found the reason: It's because [my scroll bar is set to always visible](https://www.notion.so/mui-org/Laptop-configuration-d306ca6120004d2e9034144e71c45b34?pvs=4#2bd404c6337443a0870ba1238293a457), and this ends up causing the layout shift

siriwatknp commented:
> > @zanivan I could not see the difference. Is it the dashboard template?
> 
> @siriwatknp It's happening on `Marketing-page` and `Blog`, and I found the reason: It's because [my scroll bar is set to always visible](https://www.notion.so/mui-org/Laptop-configuration-d306ca6120004d2e9034144e71c45b34?pvs=4#2bd404c6337443a0870ba1238293a457), and this ends up causing the layout shift

Got it. The way I see to improve it is to switch the header to position sticky but I think it should be done outside of this PR.

zanivan commented:
Bringing @oliviertassinari 's [feedback](https://github.com/mui/material-ui/pull/43604#pullrequestreview-2319915336) here:

---

Cool banner, but it looks like we need to continue to iterate on this:

1. Doesn't work on mobile:

https://github.com/user-attachments/assets/bad5b1de-4507-4087-a280-eabacd859d83

2. Can't really click on it:

https://github.com/user-attachments/assets/4cb59203-cd70-4a8b-aa21-7ed62a1420a5

---

siriwatknp commented:
> Bringing @oliviertassinari 's [feedback](https://github.com/mui/material-ui/pull/43604#pullrequestreview-2319915336) here:

fixed.

zanivan reviewed (commented):
I added some stray design tweaks, but I still have some remarks that need fixing before merging it:

- There's a bug with Dashboard background color:

https://github.com/user-attachments/assets/6d6a3068-99e9-4b49-9ca8-52421ddc38b1

- There's a bug with the theme `Select` where an icon appears behind the light/dark color mode toggle:

https://github.com/user-attachments/assets/ed785494-065c-4eb5-9c8f-1047e3e3f58a

- CodeSandbox isn't working, I don't know if it's just a bug here 🤔 

<img width="2278" alt="Screenshot 2024-10-08 at 09 51 23" src="https://github.com/user-attachments/assets/e023c921-8a29-40ef-b0f2-8c171f1c6b19">

- In almost all templates, when in mobile breakpoint, we end up losing the color mode toggle. Since users will likely see this in a split view, we need to have it there somehow—maybe we could draw some inspiration from the previous layouts that we had before adding the `TemplateFrame`, and bring it to CodeSandbox/Stackblitz UIs.

<img width="400" alt="Screenshot 2024-10-08 at 09 51 46" src="https://github.com/user-attachments/assets/6f22f043-3b3f-4972-bf59-6e28d0adbff0">

<img width="400" alt="Screenshot 2024-10-08 at 09 55 30" src="https://github.com/user-attachments/assets/ac8dd81e-bc66-416d-9ac7-ca62806d90c7">

- Images aren't loading, probably because we use static images from the docs. Do you have any suggestions about how we might solve this?

<img width="987" alt="Screenshot 2024-10-08 at 09 53 03" src="https://github.com/user-attachments/assets/19316eb1-96ca-4ea0-a8bf-0203c6f4ebf3">

- Sign-in, Sign-up and Sign-in-side color mode toggle could be improved, there are some bugs when clicking on it, as well as layout shift and overlapping. 

https://github.com/user-attachments/assets/efdc8dfb-bc74-478c-b51c-45b90c2379aa

siriwatknp commented:
> There's a bug with Dashboard background color:

Found the root cause, will open a separate PR but it's not related to this PR.

> There's a bug with the theme Select where an icon appears behind the light/dark color mode toggle:

I could not see it in Chrome (in any templates), what's your browser and which template?

> In almost all templates, when in mobile breakpoint, we end up losing the color mode toggle. Since users will likely see this in a split view, we need to have it there somehow—maybe we could draw some inspiration from the previous layouts that we had before adding the TemplateFrame, and bring it to CodeSandbox/Stackblitz UIs

Added the toggle to Dashboard, Marketing, and Blog mobile viewport (only show when open with Sandboxes)

> Images aren't loading, probably because we use static images from the docs. Do you have any suggestions about how we might solve this?

I fixed it by using a new `process.env.TEMPLATE_IMAGE_URL`. In dev, it will use the `/` path but when deployed it will use the deployed url.

zanivan commented:
@siriwatknp I tried to update the branch from upstream to get the changes from the light/dark background color PR #44059 to review the PR again, but noticed that doing this it also committed [170a9c9](https://github.com/mui/material-ui/pull/43708/commits/170a9c948dc299dd6b940c4d5e2fa6e7708e313f). Is this right?

zanivan commented:
> I could not see it in Chrome (in any templates), what's your browser and which template?

It's happening on Sign-in, Sign-up and Sign-in-side. I've tested on Arc (chrome), Firefox and Safari, and it happens in all of these browsers.

> For "there are some bugs when clicking on it", I think it comes from the custom styles from the theme, can you try to fix it?

Will work on it 👍

siriwatknp commented:
> There's a bug with the theme Select where an icon appears behind the light/dark color mode toggle

@zanivan found the issue, it's fixed in the latest change.

zanivan commented:
Awesome—I think we're almost there! I just have some small remarks:

1. On the custom theme, by default, we're only using the icons rounded, so we need to replace these in the StackBlitz/CodeSandbox (let me know if you want me to make this one):
```
import DarkModeIcon from '@mui/icons-material/DarkModeOutlined';
import LightModeIcon from '@mui/icons-material/LightModeOutlined';
```

2. I don't know if it's expected, so correct me if I'm wrong, but images aren't loading on both StackBlitz/CodeSandbox;
3. On the checkout template, we could add a little extra space atop, so we wouldn't have this overlapping components: <img width="566" alt="Screenshot 2024-10-14 at 08 42 21" src="https://github.com/user-attachments/assets/ac3069eb-4da4-40ae-a3ea-aec78fd463b6">

siriwatknp commented:
@zanivan All fixed. There is one caveat for the images, when you open with CodeSandbox, the image will use `https://mui.com` as a host.

zanivan reviewed (approved):
I think everything design-wise for adding StackBlitz and CodeSandbox is covered—nice work, @siriwatknp! It might be good to get an engineering review before merging, though @DiegoAndai @aarongarciah 

---

Side note: There are some design issues with the templates, but they aren’t directly related to the CodeSandbox/StackBlitz addition. I think it makes sense to address these in follow-up PRs.

oliviertassinari commented on docs/pages/material-ui/getting-started/templates/blog.js:
```diff
@@ -1,11 +1,14 @@
 import * as React from 'react';
 import AppTheme from 'docs/src/modules/components/AppTheme';
```
Could we rename this? We have two `AppTheme` in the React tree, and this component has nothing to do with theming.

oliviertassinari replied:
Actually, we could add a title to those pages, this could be a better DX. I imagine we should also remove things like `description="A onepirate theme"` what's the point, the page is `<meta name="robots" content="noindex,nofollow" />` it's absurd.

siriwatknp replied:
> Could we rename this? We have two `AppTheme` in the React tree, and this component has nothing to do with theming.

Thanks for the feedback but I would do it in a separate PR.

oliviertassinari reviewed (commented):
Very cool, I love to see this.

DiegoAndai commented on docs/data/material/getting-started/templates/shared-theme/themePrimitives.ts:
```diff
@@ -395,5 +395,9 @@ export const shape = {
 };
 
 // @ts-ignore
```
Nit: is this still required?

siriwatknp replied:
Yes, it's still required.

DiegoAndai reviewed (approved):
LGTM, one small nit.

DiegoAndai commented:
@zanivan lets remember to tweet about this when it releases 🙏🏼
